src/Core/src/Platform/Android/SafeAreaHandler.cs

```diff
@@ -0,0 +1,384 @@
+            }
+
+            var crossPlatformLayout = _getCrossPlatformLayout();
+
+            // Early exit: If this view doesn't implement safe area interfaces, skip entirely
+            if (crossPlatformLayout is not ISafeAreaView2 safeAreaLayout)
+            {
+                return false;
+            }
+
+            // Early exit: Check if any edge needs safe area handling before expensive parent check
+            bool hasAnySafeAreaEdge = false;
+            for (int edge = 0; edge < 4; edge++)
+            {
+                var region = safeAreaLayout.GetSafeAreaRegionsForEdge(edge);
+                if (region != SafeAreaRegions.None && region != SafeAreaRegions.Default)
+                {
+                    hasAnySafeAreaEdge = true;
+                    break;
+                }
+            }
+
+            // If no edges need safe area, return false immediately
+            if (!hasAnySafeAreaEdge)
+            {
+                return false;
+            }
+
+            // Only now check if parent handles safe area (expensive operation)
+            if (HasSafeAreaHandlingParent())
+            {
+                return false;
+            }
+
+            // Final check for specific regions that require safe area handling
+            for (int edge = 0; edge < 4; edge++)
+            {
+                var region = safeAreaLayout.GetSafeAreaRegionsForEdge(edge);
+                if (region == SafeAreaRegions.All)
+                {
+                    _safeAreaInvalidated = true;
+                    return true;
+                }
+
+                if (SafeAreaEdges.IsSoftInput(region))
+                {
+                    _safeAreaInvalidated = true;
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public bool HasSafeAreaHandlingParent()
+        {
+            // Use cached result if hierarchy hasn't changed
+            if (!_parentHierarchyChanged && _hasSafeAreaHandlingParent.HasValue)
+            {
+                return _hasSafeAreaHandlingParent.Value;
+            }
+
+            // Perform the hierarchy check and cache the result
+            _hasSafeAreaHandlingParent = CheckSafeAreaHandlingParentInHierarchy();
+            _parentHierarchyChanged = false;
+
+            return _hasSafeAreaHandlingParent.Value;
+        }
+
+        bool CheckSafeAreaHandlingParentInHierarchy()
+        {
+            // Walk up the view hierarchy to check if any parent is handling safe area
+            var parent = _owner.Parent;
+            int maxDepth = 10; // Limit traversal depth to prevent excessive hierarchy walking
+            int currentDepth = 0;
+
+            while (parent != null && currentDepth < maxDepth)
+            {
+                currentDepth++;
+
+                // Check if parent is a ContentViewGroup or LayoutViewGroup that responds to safe area
+                if (parent is ContentViewGroup parentContentGroup)
```

Comment:
What about having an interface implemented by both `ContentViewGroup` and `LayoutViewGroup` so that we can have this `Quick check` code only once?

Reply:
I think we already have one, https://github.com/dotnet/maui/blob/main/src/Core/src/Core/ICrossPlatformLayoutBacking.cs

Reply:
@albyrock87 , I have updated the code changes. If you have any concerns please let me know